xdsl/irdl/declarative_assembly_format.py

```diff
@@ -238,22 +238,19 @@ def resolve_result_types(self, state: ParsingState, op_def: OpDef) -> None:
             zip(state.result_types, op_def.results, strict=True)
         ):
             if result_type is None:
-                result_type = state.result_types[i]
-                range_length = len(result_type) if isinstance(result_type, list) else 1
```

Comment:
The `result_type` is already known to be `None`, so couldn't possibly be a `list`

Reply:
This doesn't seem true? `result_type` was set in the line above

Reply:
where does `result_type` come from?

Reply:
Ah I see